Sources/SPTPersistentCache.m

```diff
@@ -631,26 +629,35 @@ - (SPTPersistentCacheResponse *)guardOpenFileWithPath:(NSString *)filePath
         return [[SPTPersistentCacheResponse alloc] initWithResult:SPTPersistentCacheResponseCodeNotFound error:nil record:nil];
 
     } else {
+        const int SPTPersistentCacheInvalidResult = -1;
         const int flags = (writeBack ? O_RDWR : O_RDONLY);
 
         int fd = open([filePath UTF8String], flags);
-        if (fd == -1) {
+        if (fd == SPTPersistentCacheInvalidResult) {
             const int errn = errno;
             NSString *serr = @(strerror(errn));
             [self debugOutput:@"PersistentDataCache: Error opening file:%@ , error:%@", filePath, serr];
-            NSError *error = [NSError errorWithDomain:NSPOSIXErrorDomain code:errn userInfo:@{NSLocalizedDescriptionKey: serr}];
-            return [[SPTPersistentCacheResponse alloc] initWithResult:SPTPersistentCacheResponseCodeOperationError error:error record:nil];
+            NSError *error = [NSError errorWithDomain:NSPOSIXErrorDomain
+                                                 code:errn
+                                             userInfo:@{ NSLocalizedDescriptionKey: serr }];
+            return [[SPTPersistentCacheResponse alloc] initWithResult:SPTPersistentCacheResponseCodeOperationError
+                                                                error:error
+                                                               record:nil];
         }
 
         SPTPersistentCacheResponse *response = jobBlock(fd);
 
-        fd = close(fd);
-        if (fd == -1) {
+        fd = [self.posixWrapper close:fd];
+        if (fd == SPTPersistentCacheInvalidResult) {
             const int errn = errno;
             NSString *serr = @(strerror(errn));
             [self debugOutput:@"PersistentDataCache: Error closing file:%@ , error:%@", filePath, serr];
-            NSError *error = [NSError errorWithDomain:NSPOSIXErrorDomain code:errn userInfo:@{NSLocalizedDescriptionKey: serr}];
-            return [[SPTPersistentCacheResponse alloc] initWithResult:SPTPersistentCacheResponseCodeOperationError error:error record:nil];
+            NSError *error = [NSError errorWithDomain:NSPOSIXErrorDomain
+                                                 code:errn
```

Comment:
Could we got more descriptive identifiers? `errorNumber` instead of `errn` for instance.